google/cloud/pubsublite/integration_tests/publisher_integration_test.cc

```diff
@@ -48,119 +45,105 @@ auto constexpr kThroughputCapacityMiB = 4;
 }
 
-class PublisherIntegrationTest : public testing_util::IntegrationTest {
- protected:
-  PublisherIntegrationTest()
-      : topic_prefix_{TestTopicPrefix(system_clock::now())},
-        project_id_{google::cloud::internal::GetEnv("GOOGLE_CLOUD_PROJECT")
-                        .value_or("")},
-        location_id_{
-            google::cloud::internal::GetEnv("GOOGLE_CLOUD_CPP_TEST_ZONE")
-                .value_or("")},
-        admin_connection_{MakeAdminServiceConnection(
-            google::cloud::internal::PopulateCommonOptions(
-                google::cloud::internal::PopulateGrpcOptions(
-                    Options{}.set<EndpointOption>(MakeLocation(location_id_)
-                                                      ->GetCloudRegion()
-                                                      .ToString() +
-                                                  "-pubsublite.googleapis.com"),
-                    ""),
-                /*endpoint_env_var=*/{}, /*emulator_env_var=*/{},
-                /*authority_env_var=*/{}, "pubsublite.googleapis.com"))} {
-    auto topic_id = RandomTopicName();
-
-    GarbageCollect();
-
-    CreateTopicRequest req;
-    req.set_parent("projects/" + project_id_ + "/locations/" + location_id_);
-    req.set_topic_id(topic_id);
-    req.mutable_topic()->mutable_partition_config()->set_count(3);
-    req.mutable_topic()
-        ->mutable_partition_config()
-        ->mutable_capacity()
-        ->set_publish_mib_per_sec(kThroughputCapacityMiB);
-    req.mutable_topic()
-        ->mutable_partition_config()
-        ->mutable_capacity()
-        ->set_subscribe_mib_per_sec(kThroughputCapacityMiB);
-    req.mutable_topic()->mutable_retention_config()->set_per_partition_bytes(
-        kPartitionStorage);
-    EXPECT_STATUS_OK(admin_connection_->CreateTopic(std::move(req)));
-    auto topic = Topic{project_id_, location_id_, topic_id};
-    topic_name_ = topic.FullName();
-    publisher_ = *MakePublisherConnection(
-        topic, Options{}.set<MaxBatchMessagesOption>(kMaxNumMessagesPerBatch));
+std::string RandomTopicId() {
+  auto generator = google::cloud::internal::DefaultPRNG(std::random_device{}());
+  std::size_t const max_topic_size = 70;
+  auto const topic_prefix = TestTopicPrefix(system_clock::now());
+  auto size = static_cast<int>(max_topic_size - 1 - topic_prefix.size());
+  return topic_prefix +
+         google::cloud::internal::Sample(
+             generator, size, "abcdefghijlkmnopqrstuvwxyz0123456789_-") +
+         google::cloud::internal::Sample(
+             generator, 1, "abcdefghijlkmnopqrstuvwxyz0123456789");
+}
+
+void GarbageCollect(AdminServiceClient client, std::string const& parent) {
+  auto stale_prefix =
+      parent + "/topics/" + TestTopicPrefix(system_clock::now() - hours(48));
+  auto topics = client.ListTopics(parent);
+  for (auto const& topic : topics) {
+    ASSERT_STATUS_OK(topic);
+    if (!std::regex_search(topic->name(), std::regex{kTopicRegex})) continue;
```

Comment:
nit: I think we should build the regex outside the loop?

Reply:
You're right. Fixed.